.github/workflows/publish-to-pypi.yml

```diff
@@ -8,9 +8,16 @@ jobs:
     steps:
       - uses: actions/checkout@master
       - name: Set up Python 3.7
-        uses: actions/setup-python@v1
+        uses: actions/setup-python@v2
         with:
           python-version: 3.7
+      - uses: actions/setup-node@v2
+      - name: Running unit tests
+        run: |
+          pip install -r requirements.txt
+          python -m unittest discover
+          npm install -g pyright
+          pyright nimbella/*
```

Comment:
I didn't see the GH run on this PR (there are no checks). Are they supposed to?
Also if there's an existing test coverage threshold, can we add it as the low bar and enforce it going forward/

Reply:
Good point - I've now added the option to run the CI/CD pipeline on PRs (as well as commits). It will only run the final stage in the pipeline (publish to PyPi) if commit is for a release tag. The Github Actions still won't run for this PR until this PR has been merged....